### DISCUSSION TITLE
Pdns-recursor: How calculate cache size properly?

### DISCUSSION THREAD

**Discussion by ousatov-ua** [PowerDNS/pdns#13835] [Q&A] [answered]
Hi!
Great resolver!

I have a question how to calculate cache size properly? I mean configuration contains setting about number of cached records:

```
max-cache-entries=1000000
```
I don't know how cache is organized in pdns-recursor, so on what max size of cache entry I have to multiply this number to find out the biggest amount of RAM required?

Thank you in advance!

**Comment by omoerbeek** [comment, marked as answer]
It is very much dependent on the actual usage patterns, an answer in the cache can by anything between a 10s of bytes to several kilo bytes. There's also some associated data stored (like signatures, TTL values, ECS info, auth server info), which also adds to RAM usage.
